roles/reproducer/tasks/push_code.yml

```diff
@@ -74,9 +74,28 @@
 - name: Push random code into the proper location
   vars:
     repo_base_dir: '/home/zuul/src'
-  when:
-    - not cifmw_reproducer_skip_fetch_repositories | default(false)
   block:
+    - name: Expand cifmw_reproducer_repositories to pull code from ansible controller to controller-0
+      when:
+        - cifmw_reproducer_skip_fetch_repositories | default(false)
+        - zuul is defined
+        - zuul.items is defined
+        - zuul.projects is defined
+        # check that the project is not alredy in the user defined
+        # cifmw_reproducer_repositories value
+        - repo.value.name not in _user_sources
+      vars:
+        _user_sources: "{{ cifmw_reproducer_repositories | default([]) | map(attribute='src') }}"
+        _repo_entry:
+          src: "{{ ansible_user_dir }}/{{ repo.value.src_dir | regex_replace('/$', '') }}/"
+          dest: "/home/zuul/{{ repo.value.src_dir }}"
+      ansible.builtin.set_fact:
+        cifmw_reproducer_repositories: "{{ cifmw_reproducer_repositories | default([]) + [_repo_entry] }}"
```

Comment:
afaik we can't override a provided parameter with `set_fact`. You'll need to get some "internal" fact for that one.

Reply:
ah right, it has lower precedence. I tested it with a job without any value for `cifmw_reprodcuer_repositories`, so I did not see any issue, good catch!